feat(docs): custom skill system messages (#1670)
livechat/livechat-public-docs: src/pages/messaging/references/system-activities/index.mdx

| ------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| [`ai_agents`](#ai_agents) | [`custom_skill_awaiting_feedback`](#custom_skill_awaiting_feedback) [`custom_skill_completed`](#custom_skill_completed) [`custom_skill_failed`](#custom_skill_failed) [`custom_skill_rejected`](#custom_skill_rejected) [`custom_skill_started`](#custom_skill_started) |
| [`cdp`](#cdp)             | [`store_order_was_created`](#store_order_was_created) [`store_product_added_to_cart`](#store_product_added_to_cart) [`store_product_removed_from_cart`](#store_product_removed_from_cart)                                                   |
| [`messaging`](#messaging) | [`summary`](#summary)                                                                                                                                                                                                                       |

## ai_agents

> AI Agents source includes system events related to AI Agent activities, such as custom skill execution. Custom skill system events are visible to agents.

Reviewer: Are those visible to all agents or are there any conditions?

Author: All agents, if you can see the chat you can see the system events

### custom_skill_awaiting_feedback

> Sent when a custom skill is awaiting human feedback to continue execution.

Reviewer: ```suggestion
Generated when a custom skill is awaiting human feedback to continue execution.
```

> #### Version 2

Reviewer: ```suggestion
#### Version 1
```

| Parameter   | Required | Data type | Notes                                 |
| ----------- | -------- | --------- | ------------------------------------- |
| `agentId`   | Yes      | `string`  | The ID of the AI Agent.               |
| `skillId`   | Yes      | `string`  | The ID of the executed custom skill.  |
| `skillName` | Yes      | `string`  | The name of the executed custom skill. |

<Code>
  <CodeResponse
    title={"Sample system event payload"}
    type={"messaging-references"}
    json={"customSkillAwaitingFeedbackV2"}
  />
</Code>

### custom_skill_completed

> Sent when a custom skill has completed its execution successfully.

Reviewer: ```suggestion
Generated when a custom skill has completed its execution successfully.
```

> #### Version 2

Reviewer: ```suggestion
#### Version 1
```

| Parameter   | Required | Data type | Notes                                 |
| ----------- | -------- | --------- | ------------------------------------- |
| `agentId`   | Yes      | `string`  | The ID of the AI Agent.               |
| `skillId`   | Yes      | `string`  | The ID of the executed custom skill.  |
| `skillName` | Yes      | `string`  | The name of the executed custom skill. |

<Code>
  <CodeResponse
    title={"Sample system event payload"}
    type={"messaging-references"}
    json={"customSkillCompletedV2"}
  />
</Code>

### custom_skill_failed

> Sent when a custom skill execution has failed.

Reviewer: ```suggestion
Generated when a custom skill execution has failed.
```

> #### Version 2

Reviewer: ```suggestion
#### Version 1
```

| Parameter   | Required | Data type | Notes                                 |
| ----------- | -------- | --------- | ------------------------------------- |
| `agentId`   | Yes      | `string`  | The ID of the AI Agent.               |
| `skillId`   | Yes      | `string`  | The ID of the executed custom skill.  |
| `skillName` | Yes      | `string`  | The name of the executed custom skill. |
| `error`     | Yes      | `string`  | The error that occurred during execution. |
| `reason`    | Yes      | `string`  | The reason why the execution failed.  |

<Code>
  <CodeResponse
    title={"Sample system event payload"}
    type={"messaging-references"}
    json={"customSkillFailedV2"}
  />
</Code>

### custom_skill_rejected

> Sent when a custom skill execution has been rejected.

Reviewer: ```suggestion
Generated when a custom skill execution has been rejected by an agent.
```

Reviewer: I assume this is in response to awaiting feedback and generates when it's rejected?

Author: Correct, this is likely not possible to be triggered given current processing logic but it is part of the API

Reviewer: Ok then let's appply this suggestion 😌

> #### Version 2

Reviewer: ```suggestion
#### Version 1
```

| Parameter   | Required | Data type | Notes                                 |
| ----------- | -------- | --------- | ------------------------------------- |
| `agentId`   | Yes      | `string`  | The ID of the AI Agent.               |
| `skillId`   | Yes      | `string`  | The ID of the executed custom skill.  |
| `skillName` | Yes      | `string`  | The name of the executed custom skill. |

<Code>
  <CodeResponse
    title={"Sample system event payload"}
    type={"messaging-references"}
    json={"customSkillRejectedV2"}
  />
</Code>

### custom_skill_started

> Sent when a custom skill execution has begun.

Reviewer: ```suggestion
Generated when a custom skill execution has started.
```

> #### Version 2

Reviewer: ```suggestion
#### Version 1
```
